firmware/esp32-s3-arduino/src/main.cpp

```diff
@@ -74,36 +105,50 @@ void loop()
   // }
 
   // TODO: Validate Estop Functionality
-  if (digitalRead(ESTOP) == HIGH)
+  if (digitalRead(GPIO_NUM_7) == LOW)
   {
-    current_state = FSM_BOOT;
-    next_state = FSM_STANDBY;
-
-    pwm_set_motor_left_freq(0);
-    pwm_set_motor_right_freq(0);
-
-    ledc_timer_pause(LEDC_MODE, LEDC_TIMER_MOTOR_LEFT);
-    ledc_timer_pause(LEDC_MODE, LEDC_TIMER_MOTOR_RIGHT);
-
-    // pwm_clear_duty_pct();
-
-    digitalWrite(DEBUG_LED, LOW);
-    digitalWrite(LAMP1_ON, LOW);
-    digitalWrite(LAMP2_ON, LOW);
-    digitalWrite(LAMP3_ON, LOW);
-    digitalWrite(LAMP4_ON, LOW);
-
-    digitalWrite(GPIO_NUM_39, LOW);
     digitalWrite(GPIO_NUM_16, LOW);
-
-    twai_clear_receive_queue();
+    delay(100);
+    digitalWrite(GPIO_NUM_16, HIGH);
+    if (kick_count_1 == 0)
+    {
+      kick_time_1 = millis();
+    }
+    kick_count_1++;
+  }
+  else if (digitalRead(GPIO_NUM_37) == LOW)
+  {
+    digitalWrite(GPIO_NUM_39, LOW);
+    delay(100);
+    digitalWrite(GPIO_NUM_39, HIGH);
+    if (kick_count_2 == 0)
+    {
+      kick_time_2 = millis();
+    }
+    kick_count_2++;
+  }
+  else if ((kick_count_1 >= 2) || (kick_count_2 >= 2) )
```

Comment:
We're gonna need to talk through this logic in the morning!!

From what I'm looking at, if either of these counters equal or exceed two, fire the estop. Kick counts increment when their respective alarm pins go low in the main loop(). Although kick counts only reset if the time since their last kick increment has been greater than a minute, so I'm pretty sure with this logic we're going to be perpetually in the boot state as a result of spamming the estop